### PR TITLE
[Jobs] Add minimal long-running release test with many jobs

### LINKED ISSUE #28090: [Jobs] Add load test with high number of jobs on long running Ray cluster
### Description

Add a test with a long-running Ray cluster and a high number (say 200+) of jobs being submitted, to detect issues such as https://github.com/ray-project/ray/issues/28058.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds the simplest possible long-running release test for Ray job submission.  Submits many jobs over the course of 8 hours.  The jobs don't use `ray.init()` or runtime environments.

## Related issue number

Closes https://github.com/ray-project/ray/issues/28090
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
@krfricke sorry to tag you on every new release test, do you mind just checking the the YAML part? (the schema looks different from last time I looked at it, so I copied from another recent long running test)

File: release/jobs_tests/workloads/jobs_long_running.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    start = time.time()
+
+    address = os.environ.get("RAY_ADDRESS")
+    job_name = os.environ.get("RAY_JOB_NAME", "jobs_basic")
+
+    if address is None or not address.startswith("anyscale://"):
+        address = "http://127.0.0.1:8265"
+
+    client = JobSubmissionClient(address)
```
Comment (sihanwang41) [inline release/jobs_tests/workloads/jobs_long_running.py:89]: do we support multiple clients?

File: release/jobs_tests/workloads/jobs_long_running.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        job_ids.append(job_id)
+        print(f"submitted job: {job_id}")
+
+    for job_id in job_ids:
+        status = wait_until_finish(client, job_id, timeout_s, retry_interval_s)
+        if status != JobStatus.SUCCEEDED:
+            print(
+                f"job {job_id} failed with status {status} (`None` indicates timeout)"
+            )
+            return False
```
Comment (simon-mo) [inline release/jobs_tests/workloads/jobs_long_running.py:63]: should these happen concurrently to simulate real workload? (use ray or threading libraries)?

File: release/jobs_tests/workloads/jobs_long_running.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    client: JobSubmissionClient,
+    num_jobs: int,
+    timeout_s: int = 10 * 60,
+    retry_interval_s: int = 1,
+) -> bool:
+    job_ids = []
+    for i in range(num_jobs):
+        job_id = client.submit_job(
+            runtime_env={"working_dir": os.path.dirname(os.path.abspath(__file__))},
+            entrypoint="echo starting && sleep 60 && echo finished",
```
Comment (simon-mo) [inline release/jobs_tests/workloads/jobs_long_running.py:52]: can we do a simple ray task? (so it's more realistic)

**Review by simon-mo — APPROVED** [review]
mostly concerned about the workload being too simplistic, left two suggestions!

**Comment by krfricke** [comment]
Sorry I've missed this PR. Yaml looks good to me, let me know if you need any further support.

**Comment by architkulkarni** [comment]
Linkcheck unrelated (no docs changes)
Windows test_cancel flaky on master
Windows test_metrics_agent unrelated (this PR just adds a Python release test)

**Comment by architkulkarni** [comment]
Will follow up with more realistic workloads (`ray.init()`, GPUs, multinode with resource requests)

**Pull request merged**